### PR TITLE
ITSADSSD-26058 Forms uplift part one

### PR DESCRIPTION
**Pull request by uqmchu**
Still a work in progress. Causing some things to regress. But there's a lot to review already.

## Visual regressions to be fixed:
- [x] header search
- [x] form validation docs
- [x] working example: password form

### CODE REVIEW

**Review by Bcrunk — COMMENTED** [review]
I would have expected the disabled and Read-only styles to be stronger, typically with a Grey background.

File: packages/storybook-html/src/patterns/form-validation/examples/password-reset.scss [hunk 1]
```diff
@@ -108,7 +108,7 @@
   }
 }
 
-.form__inline-error-message {
+.inline-error-message {
```
Comment (Bcrunk) [inline packages/storybook-html/src/patterns/form-validation/examples/password-reset.scss:111]: Should this not be a modifier class 

.error-summary
.error-summary--inline

Comment (Bcrunk) [reply packages/storybook-html/src/patterns/form-validation/examples/password-reset.scss:111]: Sorry just realized  error-summary is the overall report not the line item. 

Still think you might consider a name change as the inline is implied as a changeable attribute 

.error-message
.error-message--inline

.error
.error--inline

.input-error
.input-error--inline

**Review by Bcrunk — APPROVED** [review]

**Comment by uqmchu** [comment]
Update radio button and checkbox styles

File: packages/form/src/scss/_global.scss [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
-// Forms
-$form-required-color: core.$uq-red-dark20; //d6492a;
-$form-description-color: core.$uq-neutral-dark60; //666;
-$form-input-border-color: core.$uq-neutral-dark20; //ccc;
+  $form-border-color-focus: core.$blue;
+  $form-border-color: core.$grey-10;
+  $form-border-radius: 2px;
+  $form-border-width: 1px;
+  $form-color-error: core.$alert;
+  $form-color-success: darken(core.$green, 10%);
```
[the remark this answers is not in the document]
Comment (uqmchu) [reply packages/form/src/scss/_global.scss]: Core updated w/ docs

File: packages/storybook-html/src/patterns/form-validation/examples/password-reset.scss (hunk 1 above)
Comment (uqmchu) [reply packages/storybook-html/src/patterns/form-validation/examples/password-reset.scss:111]: 'Inline' isn't a variant but aligns with existing form validation vernacular. I.e. error summary (outside the form contents) vs. inline  messages (within the form contents). I will change to `error-message` just to simplify anyway.

**Pull request merged**